source/base/mpi.cc

```diff
@@ -243,6 +243,22 @@ namespace Utilities
       return res;
     }
 
+    std::vector<IndexSet>
+    create_evenly_distributed_partitioning(
+      const MPI_Comm &           comm,
+      const IndexSet::size_type &total_size)
```

Comment:
I am not sure about the interface of this function. Passing a communicator seems not necessary. It is enough to pass `n_partitions`:
```suggestion
    std::vector<IndexSet>
    create_evenly_distributed_partitioning(
      const unsigned int         n_partitions,
      const IndexSet::size_type &total_size)
```

Reply:
For simplicity, I made it call create_ascending_partitioning(), which is a collective operation and requires a communicator. However, it's true that I can make it completely independent of MPI calls.

I think it still makes sense to put it in the Utilities::MPI, next to create_ascending_partitioning() since this partitioning will only be needed for parallel computations.

Reply:
Come to think of it, whatever function is calling this one will have to call Utilities::MPI::n_mpi_processes(MPI_Comm) beforehand to get n_partitions. Might as well put it within this function by passing the communicator

I can still make the function non-collective by removing its dependence on create_ascending_partition() if that's preferable.

Reply:
Yes, I think passing `MPI_Comm` is fine even if just for getting the local id and the number of total partitions.
I would prefer to not call `create_ascending_partitioning()`, though.

Reply:
I would prefer to have two functions (name?): one is taking a communicator, calls `Utilities::MPI::n_mpi_processes(comm)` and calls the other function.

Sounds good:

> I can still make the function non-collective by removing its dependence on create_ascending_partition() if that's preferable.

Reply:
Thanks. Made the changes as suggested and force-pushed them.